lib/ain-ocean/src/api/rawtx.rs

```diff
@@ -1,27 +1,148 @@
+        Ok(mempool_tx) => {
+            let results = mempool_tx
+                .into_iter()
+                .map(|tx_result| MempoolAcceptResult {
+                    txid: tx_result.txid,
+                    allowed: tx_result.allowed,
+                    reject_reason: tx_result.reject_reason,
+                    vsize: tx_result.vsize,
+                    fees: tx_result.fees.map(|f| f.base),
+                })
+                .collect::<Vec<MempoolAcceptResult>>();
+            Ok(Response::new(results))
+        }
+        Err(e) => {
+            eprintln!("Failed to send raw transaction: {:?}", e);
+            Err(Error::RpcError(e))
+        }
+    }
 }
 
-async fn test_rawtx() -> String {
-    "Testing raw transaction".to_string()
+#[ocean_endpoint]
+async fn get_raw_tx(
+    Path((txid, verbose)): Path<(String, bool)>,
+    Extension(ctx): Extension<Arc<AppContext>>,
+) -> Result<Response<RawTransaction>> {
+    let tx_hash = Txid::from_str(&txid)?;
+    let tx_result = ctx.client.get_transaction(&tx_hash, Some(verbose)).await?;
+    let details: Vec<TransctionDetails> = tx_result
+        .details
+        .into_iter()
+        .map(|detail| TransctionDetails {
+            address: detail.address,
+            category: detail.category.to_owned(),
+            amount: detail.amount.to_sat(),
+            label: detail.label,
+            vout: detail.vout,
+            fee: detail.fee.map(|f| f.to_sat()),
+            abandoned: detail.abandoned,
+            hex: hex::encode(&tx_result.hex),
+            blockhash: tx_result.info.blockhash,
+            confirmations: tx_result.info.confirmations,
+            time: tx_result.info.time,
+            blocktime: tx_result.info.blocktime,
+        })
+        .collect();
+
+    let bip_125 = match tx_result.info.bip125_replaceable {
+        Bip125Replaceable::Yes => Some("Yes".to_string()),
+        Bip125Replaceable::No => Some("No".to_string()),
+        Bip125Replaceable::Unknown => Some("Unknown".to_string()),
+    };
+
+    let raw_tx = RawTransaction {
+        info: WalletTxInfo {
+            confirmations: tx_result.info.confirmations,
+            blockhash: tx_result.info.blockhash,
+            blockindex: tx_result.info.blockindex,
+            blocktime: tx_result.info.blocktime,
+            blockheight: tx_result.info.blockheight,
+            txid: tx_result.info.txid,
+            time: tx_result.info.time,
+            timereceived: tx_result.info.timereceived,
+            bip125_replaceable: bip_125,
+            wallet_conflicts: tx_result.info.wallet_conflicts,
+        },
+        amount: tx_result.amount.to_sat(),
+        fee: tx_result.fee.map(|amount| amount.to_sat()),
+        details,
+        hex: tx_result.hex,
+    };
+    Ok(Response::new(raw_tx))
 }
 
-async fn get_rawtx(Path(txid): Path<String>) -> String {
-    format!("Details of raw transaction with txid {}", txid)
+fn validate(hex: String) -> Result<()> {
```

Comment:
missing dftx and composite swap tx checks
```
if ((transaction.vout[0].script.stack[1] as OP_DEFI_TX).tx.type !== CCompositeSwap.OP_CODE) {
      return
    }

    const dftx = (transaction.vout[0].script.stack[1] as OP_DEFI_TX).tx.data as CompositeSwap
    if (dftx.pools.length === 0) {
      return
    }

    const lastPoolId = dftx.pools[dftx.pools.length - 1].id
    const toTokenId = `${dftx.poolSwap.toTokenId}`

    const info = await this.deFiDCache.getPoolPairInfo(`${lastPoolId}`)
    if (info?.idTokenA === toTokenId || info?.idTokenB === toTokenId) {
      return
    }
    throw new BadRequestApiException('Invalid CompositeSwap')
```

Reply:
Issue is deserialising the `hex` string to `defichain_rpc::defichain_rpc_json::{blockchain::Transaction}`. Is there any implementation ?

Reply:
try refer
https://github.com/DeFiCh/ain/blob/e3cb077a2e51210e93618ba6d92f5b873e523901/lib/ain-ocean/src/indexer/mod.rs#L146-L177